matminer/datasets/dataset_retrieval.py

```diff
@@ -90,26 +85,122 @@ def available_datasets(print_datasets=True, print_descriptions=True,
 
     Returns: (list)
     """
-    dataset_dict = _load_dataset_dict()
+    global _dataset_dict
+
+    if _dataset_dict is None:
+        _dataset_dict = _load_dataset_dict()
 
     if sort_method not in {"alphabetical", "num_entries"}:
         raise ValueError("Error, unsupported sorting metric {}"
                          " see docs for options".format(sort_method))
 
     if sort_method == 'num_entries':
-        dataset_names = sorted(dataset_dict.keys(),
-                               key=lambda x: dataset_dict[x]["num_entries"],
+        dataset_names = sorted(_dataset_dict.keys(),
+                               key=lambda x: _dataset_dict[x]["num_entries"],
                                reverse=True)
     else:
-        dataset_names = sorted(dataset_dict.keys())
+        dataset_names = sorted(_dataset_dict.keys())
 
     # If checks done before for loop to avoid unnecessary repetitive evaluation
     if print_datasets and print_descriptions:
         for name in dataset_names:
             # Printing blank line with sep=\n to give extra line break
-            print(name, dataset_dict[name]["description"], "", sep="\n")
+            print(name, _dataset_dict[name]["description"], "", sep="\n")
     elif print_datasets:
         for name in dataset_names:
             print(name)
 
-    return list(dataset_names)
+    return dataset_names
+
+
+def get_dataset_attribute(dataset_name, attrib_key):
```

Comment:
Is there a utility function for just getting the whole dataset description?

Reply:
Yes, get_dataset_description(dataset_name) returns a human readable string description of the dataset.

Reply:
Oops, I must have overlooked it.